### PR TITLE
More tests

### PR DESCRIPTION
**Pull request by MajorBreakfast**

### CODE REVIEW

File: futures-channel/tests/mpsc.rs [hunk 1]
```diff
@@ -410,7 +410,7 @@ async fn stress_poll_ready_sender(mut sender: mpsc::Sender<u32>, count: u32) {
     for i in (1..=count).rev() {
         // FIXME: should assert `is_ok()`, but cannot because assigning the result
         // of this expression to a variable causes an ICE. See rust-lang/rust/issues/51995.
-        await!(sender.send(i));
+        await!(sender.send(i)).unwrap();
```
Comment (cramertj) [inline futures-channel/tests/mpsc.rs]: huh-- interesting that this works.

Comment (MajorBreakfast) [reply futures-channel/tests/mpsc.rs]: It returns a `Result` which throws a warning if we don't handle it

Comment (cramertj) [reply futures-channel/tests/mpsc.rs]: @MajorBreakfast Yes-- I meant that it's interesting that `unwrap()` doesn't cause the same invalid MIR error that `.is_ok()` does.

Comment (cramertj) [reply futures-channel/tests/mpsc.rs]: (see the linked issue)

Comment (MajorBreakfast) [reply futures-channel/tests/mpsc.rs]: Haha I didn‘t even read that xD All I saw was a pesky warning

**Pull request merged**